### PR TITLE
RESTful method

### PR DESCRIPTION
**Pull request by cheekujodhpur**
Modelled using representational state transfer.

Only skeleton structure, is not complete yet

### CODE REVIEW

**Comment by ranveeraggarwal** [comment]
This is great 😃 
I guess the future problems should not concentrate on implementation details and more on the problem itself. We might get more such solutions then. 
@cheekujodhpur can you add some description for newbies in the README?
Also, rename the project dir from `soln-restful` to `soln-python-restful`.

**Comment by cheekujodhpur** [comment]
Renamed the solution folder, then add and remove implemented.

Test it out and tell me more. This is very complicated to code in a good manner, so I want you to take a look before I proceed.

File: rattlesnake/01-a-class-of-students/soln-python-restful/main.py [hunk 1]
```diff
@@ -0,0 +1,181 @@
+#!/usr/bin/env python
+
+import re,sys
```
Comment (DheerendraRathor) [inline rattlesnake/01-a-class-of-students/soln-python-restful/main.py:3]: Global imports on different lines

File: rattlesnake/01-a-class-of-students/soln-python-restful/README.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+**Author**: [Kumar Ayush](http://github.com/cheekujodhpur)
+
+### RESTful
```
Comment (DheerendraRathor) [inline rattlesnake/01-a-class-of-students/soln-python-restful/README.md:3]: Mention python version to execute file

File: rattlesnake/01-a-class-of-students/soln-python-restful/main.py [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+    return rep, links
+
+def quit_handler(args):
+    for filename in data:
+        with open(filename,"w") as f:
+            for row in data[filename]:
+                f.write(','.join(row))
+                f.write("\n")
+
+    sys.exit("Goodbye m'lord!")
```
Comment (DheerendraRathor) [inline rattlesnake/01-a-class-of-students/soln-python-restful/main.py:28]: sys.exit with anything other than status code 0, results into failure. So this one will just print "Goodby..." and exit with status code 1.

File: rattlesnake/01-a-class-of-students/soln-python-restful/main.py [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+    def add_to_data(new_data):
+        for key in data:
+            data[key].append(new_data[0].split(','))
+
+    if args == None:
+        return error_state()
+    new_data = args
+    try:
+        add_to_data(new_data)
+    except:
```
Comment (DheerendraRathor) [inline rattlesnake/01-a-class-of-students/soln-python-restful/main.py:45]: As a rule of thumb, add exception type if you know what exception might be raised. For e.g. here I can see KeyError and IndexError (and Possibly AttributeError).

File: rattlesnake/01-a-class-of-students/soln-python-restful/main.py [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+
+def upload_post_handler(args):
+    def create_data(filename):
+        if filename in data:
+            return
+        with open(filename) as f:
+            lines = f.readlines()
+            student_list = filter(lambda z: z!= [], \
+                    map(lambda y: filter(lambda x:x!='',y), \
+                    map(lambda x: re.split("[^a-zA-Z0-9.]+", x), \
```
Comment (DheerendraRathor) [inline rattlesnake/01-a-class-of-students/soln-python-restful/main.py:81]: compile pattern beforehand

**Comment by DheerendraRathor** [comment]
@cheekujodhpur If possible, use an IDE like pycharm. They're really helpful in maintaining clean and readable code.

**Comment by ranveeraggarwal** [comment]
I am merging this for now, since we are going to release the next problem soon.
@cheekujodhpur please work on the points mentioned by @DheerendraRathor.

**Pull request merged**